build-canal-resource.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+set -eux
+
+# This script will download the resources associated with each charm that
+# canal cares about (ie, flannel and calico). By default, it will pull charm
+# resources from the edge channel. Call it with edge|beta|candidate|stable as
+# the first arg if you to specify the channel.
+channel=${1:-}
+if [ -z ${channel} ]; then
+  channel="edge"
+fi
```

Comment:
I have mixed feelings about this script. It seems confusing - I have to build _and push_ the flannel and calico resources to their respective charms before I run this.

I guess the concern goes like this. Suppose our automation actually built and attached resources by calling `build-calico-resource.sh` for calico and `build-canal-resource.sh` for canal. If I came along and changed the calico version in layer-calico's `build-calico-resource.sh`, then queued up new builds for canal and calico, what would happen? Would canal get the new updates? Depends which build runs first.

I dunno. It's easy enough to work with when you know what's going on, but it seems like the kind of thing that could easily be misunderstood and lead to mistakes later.

I don't really have any good alternatives to suggest. One thing that might help is to rename this to `fetch-resources-from-charm-store.sh` or something so it's clear that that's what it does. Another option might be to have this script build the resources the same way flannel or calico would by fetching the build scripts from github and running them - the source-to-source dependency seems less confusing than the source-to-build-artifact dependency, if that makes sense.